src/main.rs

```diff
@@ -498,8 +509,11 @@ struct GatewayConfig {
     #[argh(option, default = "String::from(\"confirmed\")")]
     commitment: String,
     /// default sub_account_id to use (default: 0)
-    #[argh(option, default = "0")]
-    default_sub_account_id: u16,
+    #[argh(option)]
+    default_sub_account_id: Option<u16>,
```

Comment:
not sure I see the need to change this

Reply:
I was using this to determine whether the user provided one or the other, and if I had default_sub_account_id defaulting to 0 as it was, you couldn't determine if the user actually provided it. But with your new recommended logic for combining IDs I think we can implement this...

Reply:
implemented new logic and got rid of option wrapping